src/engine/LoadURL.cpp

```diff
@@ -0,0 +1,147 @@
+string LoadURL::getCacheKeyImpl() const {
+  // TODO<qup42> do caching based on ETag, Last-Modified or similar
+  if (RuntimeParameters().get<"cache-load-results">()) {
+    return absl::StrCat("LOAD URL ", loadURLClause_.url_.asString(),
+                        loadURLClause_.silent_ ? " SILENT" : "");
+  }
+  return absl::StrCat("LOAD URL ", cacheBreaker_);
+}
+
+// _____________________________________________________________________________
+string LoadURL::getDescriptor() const {
+  return absl::StrCat("LOAD URL ", loadURLClause_.url_.asString());
+}
+
+// _____________________________________________________________________________
+size_t LoadURL::getResultWidth() const { return 3; }
+
+// _____________________________________________________________________________
+size_t LoadURL::getCostEstimate() {
+  // TODO: For now, we don't have any information about the cost at query
+  // planning time, so we just return ten times the estimated size.
+  return 10 * getSizeEstimateBeforeLimit();
+}
+
+// _____________________________________________________________________________
+uint64_t LoadURL::getSizeEstimateBeforeLimit() {
+  // TODO: For now, we don't have any information about the result size at
+  // query planning time, so we just return `100'000`.
+  return 100'000;
+}
+
+// _____________________________________________________________________________
+float LoadURL::getMultiplicity(size_t) {
+  // TODO: For now, we don't have any information about the multiplicities at
+  // query planning time, so we just return `1` for each column.
+  return 1;
+}
+
+// _____________________________________________________________________________
+bool LoadURL::knownEmptyResult() { return false; }
+
+// _____________________________________________________________________________
+std::unique_ptr<Operation> LoadURL::cloneImpl() const {
+  auto load = std::make_unique<LoadURL>(_executionContext, loadURLClause_);
+  load->cacheBreaker_ = cacheBreaker_;
+  return load;
+}
+
+// _____________________________________________________________________________
+vector<ColumnIndex> LoadURL::resultSortedOn() const { return {}; }
+
+// _____________________________________________________________________________
+Result LoadURL::computeResult(bool) {
+  // TODO<qup42> implement lazy loading; requires modifications to the parser
+  LOG(INFO) << "Loading RDF dataset from " << loadURLClause_.url_.asString()
+            << std::endl;
+  HttpOrHttpsResponse response =
+      getResultFunction_(loadURLClause_.url_, cancellationHandle_,
+                         boost::beast::http::verb::get, "", "", "");
+
+  auto throwErrorWithContext = [this, &response](std::string_view sv) {
+    std::string ctx;
+    ctx.reserve(100);
+    for (const auto& bytes : std::move(response.body_)) {
+      ctx += std::string(reinterpret_cast<const char*>(bytes.data()),
+                         bytes.size());
```

Comment:
No need for the temporary string (use a string_view for the casting, if `+=` doesn't work, then append does.

And also only copy the correct amount of bytes (make a  constant for the 100, and make something like `min(bytes.size(), maxSize - ctx.size())` the length of the appended thing.